windows/CodePush.cs

```diff
@@ -0,0 +1,668 @@
+                {
+                    this.codePushNativeModule = codePushNativeModule;
+                }
+
+                public void OnDestroy()
+                {
+                }
+
+                public void OnResume()
+                {
+                    if (lastPausedDate != null)
+                    {
+                        // Determine how long the app was in the background and ensure
+                        // that it meets the minimum duration amount of time.
+                        double durationInBackground = (new DateTime() - (DateTime)lastPausedDate).TotalSeconds;
+                        if (durationInBackground >= codePushNativeModule.minimumBackgroundDuration)
+                        {
+                            Action loadBundleAction = async () =>
+                            {
+                                await codePushNativeModule.LoadBundle();
+                            };
+
+                            codePushNativeModule.Context.RunOnNativeModulesQueueThread(loadBundleAction);
+                        }
+                    }
+                }
+
+                public void OnSuspend()
+                {
+                    // Save the current time so that when the app is later
+                    // resumed, we can detect how long it was in the background.
+                    lastPausedDate = new DateTime();
+                }
+            }
+
+            private CodePushModule codePush;
+
+            public CodePushNativeModule(ReactContext reactContext, CodePushModule codePush) : base(reactContext)
+            {
+                this.reactContext = reactContext;
+                this.codePush = codePush;
+            }
+
+            public override string Name
+            {
+                get
+                {
+                    return "CodePush";
+                }
+            }
+
+            public override IReadOnlyDictionary<string, object> Constants
+            {
+                get
+                {
+                    Dictionary<string, object> constants = new Dictionary<string, object>();
+                    constants["codePushInstallModeImmediate"] = CodePushInstallMode.IMMEDIATE;
+                    constants["codePushInstallModeOnNextRestart"] = CodePushInstallMode.ON_NEXT_RESTART;
+                    constants["codePushInstallModeOnNextResume"] = CodePushInstallMode.ON_NEXT_RESUME;
+                    constants["codePushUpdateStateRunning"] = CodePushUpdateState.RUNNING;
+                    constants["codePushUpdateStatePending"] = CodePushUpdateState.PENDING;
+                    constants["codePushUpdateStateLatest"] = CodePushUpdateState.LATEST;
+                    return constants;
+                }
+            }
+
+            public override void Initialize()
+            {
+                codePush.InitializeUpdateAfterRestart();
+            }
+
+            private async Task LoadBundle()
```

Comment:
But `codePush.GetJavaScriptBundleFileAsync(codePush.assetsBundleFileName);` is a `Task`, so this method needs to be `async` in order to await on that.